apps/openmw/mwmechanics/aipackage.cpp

```diff
@@ -402,3 +413,13 @@ DetourNavigator::Flags MWMechanics::AiPackage::getNavigatorFlags(const MWWorld::
 
     return result;
 }
+
+bool MWMechanics::AiPackage::isDoorOnTheWay(const MWWorld::Ptr& actor, const MWWorld::Ptr& door) const
+{
+    const auto world = MWBase::Environment::get().getWorld();
+    const auto halfExtents = world->getHalfExtents(actor);
+    const auto position = actor.getRefData().getPosition().asVec3() + osg::Vec3f(0, 0, halfExtents.z());
+    const auto destination = mPathFinder.getPath().front() + osg::Vec3f(0, 0, halfExtents.z());
```

Comment:
front() will cause undefined behavior if the path is empty, maybe that should be fixed to avoid issues if someone tries to call isDoorOnTheWay elsewhere (even though that's unlikely).

Reply:
Thanks for noticing but I intetionally made it that way, because there is no reason other than that to make logic of this function more complex and as you noticed there is no UB until this method will be called somewhere else. Do you have any proposals how to fix it? One way is to put `if` at the beginning and either return false or throw an exception. Exception will not help much, because function is just wrongly used when it throws. `assert` may help. At least debug builds will report an error. Don't access mPathFinder from here, pass front point instead as argument. Move this code to `openDoors` function to make it impossible to call. Which way do you prefer or maybe there is another one?